aml-development/ozp-backend
# Elasticsearch User Based Recommendation changes

Please review:
@mannyrivera2010 
@skhtet 
@J-Fid 
@rkenyon1969 
@blynch11p 

The changes are for implementation for the Elasticsearch User Based Recommendation changes.  The changes also include changes from ozpcenter/api/storefront/model_access.py which are being reviewed by another PR #301.  If needed can remove the changes before merging, but needed to see changes in listings.  

**Note**: The bookmarked listings do not get returned after running the make recommend command for Elasticsearch.  A similar feature can be added to all results if this is desired.

To test recommendations:
Download code
(optional) Clear ozpcenter_recommendentry database contents table using approved SQLite browser
Run: ES_ENABLED=True make recommend
(Command will run all recommendation engines.  If only elasticsearch is required, need to modify /ozp_backend/ozpcenter/scripts/recommend.py by removing the baseline and graph_cf engines when ES_ENABLED is True)
The result should show a difference in Recommendation listings.  To compare one can run without Elasticsearch by running:
make recommend
(Without the ES_ENABLED)
